hail/src/main/scala/is/hail/linalg/GridPartitioner.scala

```diff
@@ -111,43 +112,46 @@ case class GridPartitioner(blockSize: Int, nRows: Long, nCols: Long, maybeBlocks
 
   /**
     *
-    * @return A transposed GridPartitioner and a function that maps partitions in the new partitioner to partitions
-    *         in the old partitioner. 
+    * @return A transposed GridPartitioner and a function that maps partitions in the new transposed partitioner to
+    *         the parent partitions in the old partitioner.
     */
   def transpose: (GridPartitioner, Int => Int) = {
     val gpT = GridPartitioner(blockSize, nCols, nRows)
-    def transposeBI(bi: Int): Int = gpT.coordinatesBlock(this.blockBlockCol(bi), this.blockBlockRow(bi))
-    def inverseTransposeBI(bi: Int) = this.coordinatesBlock(gpT.blockBlockCol(bi), gpT.blockBlockRow(bi))
-    maybeBlocks match {
+    partitionIndexToBlockIndex match {
       case Some(bis) =>
-        val (biTranspose, piTranspose) = bis.map(transposeBI).zipWithIndex.sortBy(_._1).unzip
-        val inverseTransposePI = piTranspose.zipWithIndex.sortBy(_._1).map(_._2)
-        
-        (GridPartitioner(blockSize, nCols, nRows, Some(biTranspose)), inverseTransposePI)
-      case None => (gpT, inverseTransposeBI)
+        def transposeBI(bi: Int): Int = gpT.coordinatesBlock(this.blockBlockCol(bi), this.blockBlockRow(bi))
+
+        val (partIdxToBlockIdxT, partIdxTToPartIdx) = bis.map(transposeBI).zipWithIndex.sortBy(_._1).unzip
```

Comment:
this is `partIdxT...` It's the transposed partition index to the transposed block index.

The `partIdxToBlockIdxT` would be:
```
bis.map(transposeBi)
```